bigframes/dataframe.py

```diff
@@ -925,10 +927,53 @@ def drop(
             )
         if columns:
             block = block.drop_columns(self._sql_names(columns))
-        if not index and not columns:
+        if index is None and not columns:
             raise ValueError("Must specify 'labels' or 'index'/'columns")
         return DataFrame(block)
 
+    def _drop_by_index(self, index: indexes.Index) -> DataFrame:
```

Comment:
Can we do this with a block-level join and filter op? Implementing as a block-operation will make it easier to share logic with series. I also worry that this complex index manipulation has edge cases, in particular when it comes to column multi-indices.

Reply:
Okay, I have refactored it to use block-level join and filter op. The code should be much improved now, or at least much shortened.